bookmark-it/scripts/handle-button-click.js

```diff
@@ -0,0 +1,31 @@
+/**
+In this file:
+
+On button click:
+* find out if the tab's URL is bookmarked.
+*   if it isn't, bookmark it.
+*   if it is, remove the bookmark.
+*/
+function toggleBookmark(tabInfo) {
+
+  function updateBookmark(bookmarkArray) {
+    if (bookmarkArray.length === 0) {
+      browser.bookmarks.create({
```

Comment:
Since you're using a `catch` function down below, the returned promise should be returned to the chain:
````js
return browser.bookmarks.create({...});
````